Marlin/Marlin_main.cpp

```diff
@@ -3207,58 +3203,63 @@ static void homeaxis(const AxisEnum axis) {
       sync_plan_position_e();
       prepare_move_to_destination();
 
-      // Is a Z hop set, and has the hop not yet been done?
-      if (has_zhop) {
-        hop_amount += retract_zlift;                // Carriage is raised for retraction hop
+      // Is a Z hop set, and has the hop not yet been done? && no double zlifting if hop_amount exist
+      if (has_zhop && !hop_amount) {
```

Comment:
**No conditionnals here** 
The machine is the machine 
The gcode is the gcode
If the machine make a hop , than the machine can recover and it's all